app/controllers/api/pictures_controller.rb

```diff
@@ -1,4 +1,13 @@
 module Api
   class PicturesController < BaseController
+    def create_resource(_type, _id, data)
+      unless %(png jpg).include?(data['extension'])
+        raise BadRequestError, 'Failed to create Picture - extension must be png or jpg'
```

Comment:
If we only support specific extensions, should this validation be in the model?  If it's there, then you don't need the explicit check here and it will be handled automatically by lines 7-11